gunicorn/workers/gthread.py

```diff
@@ -206,8 +206,13 @@ def run(self):
 
             # can we accept more connections?
             if self.nr_conns < self.worker_connections:
+                # Block for new events until worker times out or keepalive timeout.
+                select_timeout = self.timeout or 1.0
+                if self._keep:
```

Comment:
I can hoist this change out of the while loop, but then folks running the default config won't see any noticeable difference. 

Keeping it inside the loop allows making timeout adaptive to the presence of current keepalived sockets.